magicgui/type_map.py

```diff
@@ -225,9 +240,90 @@ def pick_widget_type(
     return widgets.EmptyWidget, {"visible": False}
 
 
+_SIMPLE_RETURN_TYPES = [
+    bool,
+    int,
+    float,
+    str,
+    pathlib.Path,
+    datetime.time,
+    datetime.date,
+    datetime.datetime,
+    range,
+    slice,
+]
+
+
+@return_matcher
+def default_return_matcher(value, annotation) -> WidgetTuple | None:
+    """Checks for 'simple' types that fit in a LineEdit."""
+    dtype, optional = _normalize_type(value, annotation)
+    if dtype in _SIMPLE_TYPES:
+        return widgets.LineEdit, {"gui_only": True}
+    else:
+        return None
+
+
+@return_matcher
+def tabular_return_matcher(value, annotation) -> WidgetTuple | None:
+    """Checks for tabular data."""
+    # TODO: is this correct?
+    if annotation == inspect._empty:
+        return None
+    dtype, optional = _normalize_type(value, annotation)
+    args = [_evaluate_forwardref(a) for a in get_args(widgets._table.TableData)]
+    if dtype in args:
+        return widgets.Table, {}
+
+    return None
+
+
+def return_widget_type(
+    value: Any = None,
+    annotation: type | None = None,
+    options: WidgetOptions | None = None,
+) -> WidgetTuple:
+    """Pick the appropriate widget type for ``value`` with ``annotation``."""
+    options = options or {}
+    annotation = _evaluate_forwardref(annotation)
+    dtype, optional = _normalize_type(value, annotation)
+    if optional:
+        options.setdefault("nullable", True)
+    choices = options.get("choices") or (isinstance(dtype, EnumMeta) and dtype)
+
+    if "widget_type" in options:
+        widget_type = options.pop("widget_type")
+        if choices:
+            if widget_type == "RadioButton":
+                widget_type = "RadioButtons"
+                warnings.warn(
+                    f"widget_type of 'RadioButton' (with dtype {dtype}) is being "
+                    "coerced to 'RadioButtons' due to choices or Enum type.",
+                    stacklevel=2,
+                )
+            options.setdefault("choices", choices)
+        return widget_type, options
+
+    # look for subclasses
+    for registered_type in _TYPE_DEFS:
+        if dtype == registered_type or _is_subclass(dtype, registered_type):
+            _cls, opts = _TYPE_DEFS[registered_type]
+            return _cls, {**options, **opts}  # type: ignore
+
+    for matcher in _RETURN_MATCHERS:
+        _widget_type = matcher(value, annotation)
+        if _widget_type:
+            _cls, opts = _widget_type
+            return _cls, {**options, **opts}  # type: ignore
+
+    # Chosen for backwards/test compatibility
+    return widgets.LineEdit, {"gui_only": True}
+
+
 def get_widget_class(
     value: Any = None,
     annotation: type | None = None,
+    is_input: bool = None,
```

Comment:
this is technically a breaking API change, since this is a public function and this changes the order of positional args.  (not to mention the fact that the current default of `is_input=None` will change this function to return widgets by default). Can we put this after `options`, change the name to `is_result` (to match the current wording used elsewhere of `result_widget`, and use a default value of `is_result=False`?

Reply:
I made the suggested change and rebased it into the commit that added this so that we wouldn't have any commits where `is_input` existed. Please let me know what you think.